src/stores/GuildEmojiRoleStore.js

```diff
@@ -28,7 +36,7 @@ class GuildEmojiRoleStore extends DataStore {
       return Promise.reject(new TypeError('INVALID_TYPE', 'roles',
         'Array or Collection of Roles or Snowflakes', true));
     }
-    const newRoles = [...new Set(roleOrRoles.concat(this.array()))];
+    const newRoles = [...new Set(roleOrRoles.concat(this.keys()))];
```

Comment:
Isn't this now concatenating Role instances with role ids?
The point of only using Role instances in a Set was to eliminate duplicated entries to avoid a quite cryptic error from discord.